src/system.rs

```diff
@@ -1083,6 +1098,15 @@ macro_rules! system {
                     value: self.value.min(other.value),
                 }
             }
+
+            #[inline(always)]
+            fn clamp(self, min: Self, max: Self) -> Self {
```

Comment:
This implementation isn't explicitly needed. `Ord` has a default implementation. I supose it's possible a manual implementation could be performant.

https://www.rustexplorer.com/b#%2F*%0A%5Bdependencies%5D%0Auom%20%3D%20%7B%20version%20%3D%20%22*%22%2C%20features%20%3D%20%5B%22i32%22%5D%20%7D%0A*%2F%0A%0Ause%20uom%3A%3Asi%3A%3Ai32%3A%3A*%3B%0Ause%20uom%3A%3Asi%3A%3Alength%3A%3A%7Bmeter%7D%3B%0A%0Afn%20main()%20%7B%0A%20%20%20%20let%20a%20%3D%20Length%3A%3Anew%3A%3A%3Cmeter%3E(15)%3B%0A%20%20%20%20let%20b%20%3D%20a.clamp(Length%3A%3Anew%3A%3A%3Cmeter%3E(17)%2C%20Length%3A%3Anew%3A%3A%3Cmeter%3E(20))%3B%0A%0A%20%20%20%20println!(%22%7B%3A%3F%7D%22%2C%20b)%3B%0A%7D